src/main/java/com/treasure/hunt/strategy/hider/impl/GameField.java

```diff
@@ -0,0 +1,521 @@
+                bestCalc = calcOnEdge;
+                log.debug("new best Const " + bestCalc.getValue() + " at " + bestCalc.getKey() + " at line (" + edges[currentCoordinateIndex - 1] + ", " + edges[currentCoordinateIndex] + ")");
+            }
+        }
+
+        worstEdgePoints.sort(new Comparator<Pair<Coordinate, Double>>() {
+            @Override
+            public int compare(Pair<Coordinate, Double> coordinateDoublePair, Pair<Coordinate, Double> t1) {
+                return coordinateDoublePair.getValue().compareTo(t1.getValue());
+            }
+        }.reversed());
+        log.info("worst Point List:" + worstEdgePoints.size() + "first element: " + worstEdgePoints.get(0));
+        log.info("best overall Sampled constant is" + bestSampled.getValue() + " at " + bestSampled.getKey().toString());
+        log.info("best overall calculated constant is" + bestCalc.getValue() + " at " + bestCalc.getKey().toString());
+        return worstEdgePoints;
+    }
+
+    /**
+     * Alternate to {@sampleMaxConstantPointOnLineSegment}
+     * Calculates the Point on the boundarySegment described by p1,p2 of the possible Area which results in the largest Constant of PathMin * C = PathActual
+     *
+     * @param p1     the Start of the Segment v
+     * @param p2     the End of the Segment v
+     * @param player current Position of the Player
+     * @return the Point and its corresponding constant C
+     */
+    private Pair<Coordinate, Double> calcMaxConstantPointOnSegment(Coordinate p1, Coordinate p2, Coordinate player) {
+        LineSegment unscaled = new LineSegment(p1, p2);
+        //log.debug("Calculating on " + unscaled);
+        Point origin = startingPoint;
+        assert origin.equals( gf.createPoint(new Coordinate(0, 0))); // origin assumed to be (0,0)
+
+        Coordinate projection = unscaled.project(origin.getCoordinate());
+        double scalingFactor = 1.0 / projection.distance(origin.getCoordinate());
+
+        AffineTransformation scalingTransform = AffineTransformation.scaleInstance(scalingFactor, scalingFactor);
+
+        //determine rotation angle to put ProjectionPoint onto the Y-Axis
+        double rotationAngle = Angle.angleBetweenOriented(projection, origin.getCoordinate(), new Coordinate(0, 1));
+
+        AffineTransformation rotationTransform = AffineTransformation.rotationInstance(rotationAngle);
+
+        Coordinate p1Transformed = p1.copy();
+        Coordinate p2Transformed = p2.copy();
+        Coordinate playerTransformed = player.copy();
+
+        //log.trace("scaling by Factor" + scalingFactor);
+        p1Transformed = scalingTransform.transform(p1Transformed, p1Transformed);
+        p2Transformed = scalingTransform.transform(p2Transformed, p2Transformed);
+        playerTransformed = scalingTransform.transform(playerTransformed, playerTransformed);
+        //log.info("after scaling: " + new LineSegment(p1Transformed, p2Transformed));
+        //log.info(playerTransformed.toString());
```

Comment:
I am gonna stop commenting on every out commented log staement. Leave them in it is ok just trun down the level to debug or trace.